src/base/bittorrent/infohash.h

```diff
@@ -42,6 +44,15 @@ namespace BitTorrent
         InfoHash(const QString &hashString);
         InfoHash(const InfoHash &other) = default;
 
+        static constexpr int size()
```

Comment:
Maybe `underlyingSize()`?

Reply:
>Maybe underlyingSize()?

I would think it is the base class/enum size, or perhaps `underlyingHashSize()`?

Reply:
>I have mixed feelings for the name

I've thought it over again, and I don't see the problem here...
>size() sounds like the size of InfoHash but in reality it is the size of `sha1_hash`

Yes. It is.
>but in reality it is the size of `sha1_hash`

Yes. It is too. Since InfoHash is SHA1 hash (wrapper). Maybe just rename it to `SHA1Hash`?

Reply:
Since this may cause some controversy, but there is no serious problem here, we can stretch its discussion beyond this PR to not to slow it down (unless we reach consensus earlier).

Reply:
>I've thought it over again, and I don't see the problem here...

For me it is like having `Session::size()` and `Session::size() != sizeof(Session)`.
That is, the name (both `size()` & `underlyingSize()`) won't be a confusion if `InfoHash` only have one private variable `libtorrent::sha1_hash` and no others.

IMO `underlyingHashSize()` or `hashSize()` are pretty good since they are very explicit about what they represents, very low chance someone will interpret them as `sizeof(InfoHash)`.

>Since InfoHash is SHA1 hash (wrapper). Maybe just rename it to SHA1Hash?

That would be a bad idea, its logical meaning has more than its implementation.

tldr; just pick any name other than `size()` & `underlyingSize()`.

Reply:
But to have `size()` method that is identical to `sizeof(Class)` is absurd. What's the point?

>For me it is like having Session::size() and Session::size() != sizeof(Session).

Session is incorrect example here since it doesn't represent some data. InfoHash is another thing. It represents data. So having `InfoHash::size()` that means "the size of represented data" is quite correct. Otherwise, all such cases could be incorrect from your point of view (e.g. `QString::size()` etc.).

>its logical meaning has more than its implementation.

It is unclear for me... Its logical meaning is SHA1 hash.
On the contrary, the current name (InfoHash) is incorrect, because it is associated with only one specific application of this type (torrent info hash, i.e. sha1 hash of torrent info section). But it actually applies to other entities, such as pieces.

P.S. If you intend to continue this argument, pay attention to https://github.com/qbittorrent/qBittorrent/pull/10373#discussion_r266241130. Let's not slow down here.

Reply:
>Session is incorrect example here since it doesn't represent some data. 

I picked an absurd example, because using `size()` is absurd to me.
Please at least pick your own suggestion `underlyingSize()`, still better than the current and at least is suggesting that `sha1_hash` is the real underlying type not QString nor bool.

>It is unclear for me... Its logical meaning is SHA1 hash.

I picked the wrong word, I meant conceptual meaning.
By definition "info hash" is the result of hashing some data which the algorithm is chosen to be sha1 hash, it could be other hashing methods if the bittorrent committee decides to.
Of course you can still rename to SHA1Hash, because it is used as holding sha1 hash instead of hashing things, but then you'll have to change again in the future when info hash isn't sha1 anymore (although unlikely but who knows...)

Reply:
>just pick any name other than size() & underlyingSize().

Would `length` suit you?

Reply:
>Please at least pick your own suggestion underlyingSize(), still better than the current and is suggesting that sha1_hash is the real underlying type not QString nor bool.

I offered it without thinking. There's no point in indicating in the class interface that it's just a wrapper over another class. It breaks the whole point.

Reply:
>Would length suit you?

Yes!

Reply:
Ok, done!